src/include/storage/buffer_manager/buffer_managed_file_handle.h

```diff
@@ -1,10 +1,52 @@
 #pragma once
 
+#include "storage/buffer_manager/buffer_allocator.h"
 #include "storage/file_handle.h"
 
 namespace kuzu {
 namespace storage {
 
+static constexpr uint64_t LOADED_MASK = 0x8000000000000000;
+static constexpr uint64_t DIRTY_MASK = 0x4000000000000000;
+static constexpr uint64_t PAGE_IDX_MASK = 0x3FFFFFFFFFFFFFFF;
+
+enum class LockMode : uint8_t { SPIN = 0, NON_BLOCKING = 1 };
+
+class BufferManagedFileHandle;
+
+// Keeps state information for a disk page or an in memory data block.
+class PageState {
```

Comment:
I kept PageState. Instead of using the term `vmPage`, I actually fall back to `frame`. Compared to `filePage` and `vmPage`, I think `page` and `frame` is more clear, and we've built common agreement on what `frame` is, and I added brief explanation of these two terms in comment above `BufferManager`.